Upgrade versions and add extra content

radix0000 opened the pull request:
Has been upgraded as far as is possible (while sticking with sphinx-rtd-theme). Versions of important packages are:

sphinx==5.3.0
sphinx-rtd-theme==1.1.1
myst-parser==0.18.1
docutils==0.17.1

Kitchen sink and sphinx examples for tables etc. added.

Commit:
Upgrade sphinx-rtd-theme==1.1.1 etc.

Commit:
Update configuration to match theme version

Commit:
Add readthedocs yaml config file

Commit:
Fix submodule config

Commit:
Upgrade other dependencies where possible

Commit:
Include todo and autodoc Sphinx extensions

Commit:
Add Sphinx examples for tables, admonitions and roles

Commit:
Add Sphinx kitchen sink example

Commit:
Rejig contents tree

Commit:
Fix roles page

Bjwebb commented on docs/examples/admonitions.rst:
```diff
@@ -0,0 +1,176 @@
+
+It is possible to create:
+
+.. code-block:: css
+
+    :root {
+      --icon--pied-piper: url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512"><path d="M244 246c-3.2-2-6.3-2.9-10.1-2.9-6.6 0-12.6 3.2-19.3 3.7l1.7 4.9zm135.9 197.9c-19 0-64.1 9.5-79.9 19.8l6.9 45.1c35.7 6.1 70.1 3.6 106-9.8-4.8-10-23.5-55.1-33-55.1zM340.8 177c6.6 2.8 11.5 9.2 22.7 22.1 2-1.4 7.5-5.2 7.5-8.6 0-4.9-11.8-13.2-13.2-23 11.2-5.7 25.2-6 3 ...
+    }
+    .admonition.pied-piper {
+      border-color: rgb(43, 155, 70);
+    }
+    .admonition.pied-piper > .admonition-title {
+      background-color: rgba(43, 155, 70, 0.1);
+      border-color: rgb(43, 155, 70);
+    }
+    .admonition.pied-piper > .admonition-title::before {
+      background-color: rgb(43, 155, 70);
+      -webkit-mask-image: var(--icon--pied-piper);
+      mask-image: var(--icon--pied-piper);
+    }
```
This css doesn't seem to be used anywhere, so the admonition at the bottom doesn't get rendered correctly https://sphinx-base.readthedocs.io/en/upgrade_to_latest_version/examples/admonitions.html

Bjwebb commented:
I've spotted something that seems to be broken with the examples.

I'm happy with all the version updates though, so do make those same changes to [openreferral/specification](https://github.com/openreferral/specification/).

Commit:
examples/admonitions: Add missing CSS file

Add missing CSS file to format custom admonition example

https://opendataservices.plan.io/issues/36396

radix0000 commented:
Have added back missing CSS file, and custom admonition at bottom should now be rendering correctly.

Bjwebb reviewed (approved):
Approving now because admonition is now clearly styled. 

The logo looks to be slightly mangled on my screen though:
![Screenshot from 2023-01-19 17-43-46](https://user-images.githubusercontent.com/634/213520318-564464c6-fd28-4aff-a798-6646f6a8f6e3.png)

Commit:
examples/admonitions: Improve custom admonition examples

Better example of fully customised admonition, and fix to dropdown example
